### PR TITLE
New addon: snap scripts to grid

### LINKED ISSUE #3729: New addon: script snapping
### Is your request related to a problem? Please describe.

image

Placing scripts in straight lines takes a lot of time and effort, and it's unsatisfying to just... leave them not straight!

### Describe the solution you'd like

When a certain key is pressed (or perhaps by default if a certain setting is toggled) scripts would snap to certain x and y intervals on a grid when dragged or placed. It would make sense to use the workspace dots as the default grid.

### Describe alternatives you've considered

Living with unstraightened scripts for the rest of my life...

### PR DESCRIPTION
**Pull request by lisa-wolfgang**
Resolves #3729

### Changes

When this addon is enabled, scripts snap to the workspace dots in the editor:

image

The grid size of the dots can also be modified. (The screenshots use the half-block preset.)

image

### Reason for changes

script-snap Scrolling Demo

### Tests

Tested in browser w/ Chromium 95.

### CODE REVIEW

File: addons/script-snap/userscript.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export default async function ({ addon, global, console }) {
+  const Blockly = await addon.tab.traps.getBlockly();
+  const workspace = await Blockly.getMainWorkspace();
```
Comment (GarboMuffin) [inline addons/script-snap/userscript.js:3]: Haven't tried this locally but I'm guessing that this won't work if the user exits the editor and opens it again. Perhaps do something like https://github.com/ScratchAddons/ScratchAddons/blob/master/addons/editor-sounds/userscript.js

**Comment by lisa-wolfgang** [comment]
Update from W_L -- will be merged for v1.23.0.

File: addons/script-snap/addon.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "Snap scripts to grid",
+  "description": "Drag a script to automatically align its position to the workspace dots.",
```
Comment (WorldLanguages) [inline addons/script-snap/addon.json:3]: We don't ever use the word "workspace" user facing, so this should be "code area" or something else.

Comment (WorldLanguages) [reply addons/script-snap/addon.json:3]: Yeah, it's on https://scratchaddons.com/docs/reference/addon-manifest/

**Comment by Endermaster3690** [comment]
really good reason

**Pull request merged**